doc/rst/usingchapel/launcher.rst

```diff
@@ -106,6 +106,17 @@ forwarded to worker processes. However, this strategy is not always
 reliable. The remote system may override some environment variables, and
 some launchers might not correctly forward all environment variables.
 
+.. _chpl-launch-masterip:
```

Comment:
I'm not sure we should document `CHPL_LAUNCH_MASTERIP` in terms of GASNet functionality (`GASNET_MASTERIP` for example).  Now that we have the former, it should be the primary method for specifying this, and the latter is no longer needed (though it can be still be mentioned in `udp.rst`).  I think we can just say that it provides a way for program instances running on compute nodes to know where they were launched from.  (That's right, isn't it?)  Then in `udp.rst` and `aws.rst` we should document `CHPL_LAUNCH_MASTERIP` as the primary way to specify this, and say that if it's set it it provides the same information as `GASNET_MASTERIP` would.

Reply:
In the case of multilocale libraries, we're using it to specify the location where part of the program can be listening from - this will be from both the original launched location and from on the compute nodes.  I think it is useful to mention it as for the launched location when gasnet is involved, but it does seem reasonable for that not to be the primary definition of the environment variable.

Reply:
I feel like we want to do the same thing here that we've done with for example the generalized `CHPL_RT_NUM_THREADS_PER_LOCALE` versus the specialized Qthreads `QT_HWPAR`, where our own env var is the documented way to achieve the desired functionality and either we don't mention the Qthreads-specific variable at all, or only mention it in passing.  So it's just a change in phrasing: instead of saying that "`CHPL_LAUNCHER_MASTERIP` is intended as an expansion on `GASNET_MASTERIP`", document what our variable is and does, without reference to the GASNet one, and mention in our GASNet-specific documentation that our variable will set the GASNet one if the latter isn't set.  (Actually, you might consider having our variable override the GASNet one, since that's what we do elsewhere - `CHPL_RT_NUM_THREADS_PER_LOCALE` will override `QT_HWPAR` if both are set.)